File: hmwk_03.py
# Mike Urbano
# mxu2814
# 2018-09-22
#--------------------------------------------------
import re
import sys

#--------------------------------------------------
def processToken( token ) :
  ID = re.compile( '^[\\$a-z][\\dA-Z]*$|^[\\|][\\dA-Z]*[\\|]$' )
  INT = re.compile( '^[%][\\da-fA-F]+' )
  FP = re.compile( '^[\\d]+\\.[\\d]+' )
  result = ID.match(token)
  match = False
  if result:
    print( '>%s< matches ID' % ( token ) )
    match = True
  result = INT.match(token)
  if result:
    print( '>%s< matches INT' % ( token ) )
    match = True
  result = FP.match(token)
  if result:
    print( '>%s< matches FP' % ( token ) )
    match = True
  if not (match==True):
    print( '>%s< does not match' % ( token ) )

#--------------------------------------------------
def main() :
  fName = sys.argv[ 1 ]
  print( 'processing tokens from ' + fName + ' ...' )

  with open( fName, 'r' ) as fp :
    lines = fp.read().replace( '\r', '' ).split( '\n' )

  for line in lines :
    for token in line.split() :
      processToken( token )

#--------------------------------------------------
if ( __name__ == '__main__' ) :
  main()

#--------------------------------------------------

File: hmwk_03.java
// Mike Urbano
// mxu2814
// 2018-09-22

import java.nio.file.Paths;
import java.nio.file.Files;
import java.util.Arrays;
import java.util.regex.*;

public class hmwk_03 {
  public static void processToken( String token )
  {
    boolean match = false;
    Pattern ID = Pattern.compile( "^[\\$[a-z]][\\d[A-Z]]*|^[\\|][\\d[A-Z]]*[\\|]$" );
    Pattern INT = Pattern.compile( "^%[\\d[a-f[A-F]]]+" );
    Pattern FP = Pattern.compile( "[\\d]+\\.[\\d]+" );
    Matcher r = ID.matcher(token);
    boolean result = r.matches();
    if(result == true){
        System.out.println( ">" + token + "< matches ID" );
        match = true;}
    r = INT.matcher(token);
    result = r.matches();
    if(result == true){
        System.out.println( ">" + token + "< matches INT" );
        match = true;}
    r = FP.matcher(token);
    result = r.matches();
    if(result == true){
        System.out.println( ">" + token + "< matches FP" );
        match = true;}
    if(!match)System.out.println( ">" + token + "< does not match" );
  }

  public static void main( String[] args )
  {
    System.out.println( "processing tokens from " + args[ 0 ] + " ..." );

    try {
      Files.lines( Paths.get( args[ 0 ] ) ).forEachOrdered(
        line -> Arrays.stream( line.split( "\\s+" )  )
            .forEachOrdered( token -> processToken( token ) ) );
    } catch ( java.io.IOException e ) {
      e.printStackTrace();
    }
  }
}
